src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityMaintenanceHatch.java

```diff
@@ -178,177 +191,125 @@ public boolean startWithoutProblems() {
+        int index = 0;
+        while (index < stacks.size()) {
+            ItemStack stack = stacks.get(index);
+            Item item = stack.getItem();
+
+            if (item instanceof ItemGTToolbelt toolbelt) {
+                stacks.remove(index);
+                toolbelt.iterateSlots(stack, stacks::add);
+                continue;
             }
-        }
-        if (!proceed) {
-            return;
+
+            if (item.getToolClasses(stack).isEmpty()) {
+                stacks.remove(index);
+                continue;
+            }
+
+            index++;
         }
 
-        mainfor:
-        for (int i = 0; i < toolsToMatch.size(); i++) {
-            String toolToMatch = toolsToMatch.get(i);
-            if (toolToMatch != null) {
-                // Try to use the item in the player's "hand" (under the cursor)
-                ItemStack heldItem = entityPlayer.inventory.getItemStack();
-                if (heldItem.getItem() instanceof ItemGTToolbelt toolbelt) {
-                    if (toolbelt.damageAgainstMaintenanceProblem(heldItem, toolToMatch, entityPlayer)) {
-                        ((IMaintenance) getController()).setMaintenanceFixed(i);
-                        setTaped(false);
-                        continue;
-                    }
-                } else if (ToolHelper.isTool(heldItem, toolToMatch)) {
-                    fixProblemWithTool(i, heldItem, entityPlayer);
-
-                    if (toolsToMatch.stream().allMatch(Objects::isNull)) {
-                        return;
-                    }
-                    continue;
-                }
-
-                // Then try all the remaining inventory slots
-                for (ItemStack itemStack : entityPlayer.inventory.mainInventory) {
-                    if (itemStack.getItem() instanceof ItemGTToolbelt toolbelt) {
-                        if (toolbelt.damageAgainstMaintenanceProblem(itemStack, toolToMatch, entityPlayer)) {
-                            ((IMaintenance) getController()).setMaintenanceFixed(i);
-                            setTaped(false);
-                            continue mainfor;
-                        }
-                    } else if (ToolHelper.isTool(itemStack, toolToMatch)) {
-                        fixProblemWithTool(i, itemStack, entityPlayer);
-
-                        if (toolsToMatch.stream().allMatch(Objects::isNull)) {
-                            return;
-                        }
-                        continue mainfor;
-                    }
-                }
+        Set<Int2ObjectMap.Entry<String>> toolEntries = controller.getToolsForMaintenance();
+        for (ItemStack toolStack : stacks) {
+            if (toolEntries.isEmpty()) return;
+
+            Int2ObjectMap.Entry<String> entry = findMatchingClass(toolEntries,
+                    toolStack.getItem().getToolClasses(toolStack));
+            if (entry != null) {
+                ToolHelper.damageItemWhenCrafting(toolStack, player);
```

Comment:
This should play the tool sounds that are used to fix the maintenance issues, so the player knows that something happened. Although if we are fixing multiple issues at one time, it might be best to only play the sound of the first tool.